services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -91,7 +114,23 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
 
   override def getFileVersions(project: Project, path: Path)(
     implicit ec: ExecutionContext
-  ): AsyncResult[List[Version]] = ???
+  ): AsyncResult[Seq[Version]] =
+    for {
+      projectVersions <- getProjectVersions(project)
+      fileCommits <- getFileCommits(project, path)
+    } yield {
+      if (projectVersions.isRight && fileCommits.isRight) {
+        val tagsProject = projectVersions.right.get
+        val tagsFiles = fileCommits.right.get
+        Right(for {
+          tagProject <- tagsProject
+          tagFile <- tagsFiles
+          if tagFile.commit_id == tagProject.commit.id
+        } yield tagProject)
+      } else {
+        Left(VersioningException(fileCommits.left.get.message))
+      }
+    }
```

Comment:
Use pattern matching for this, call right.get or left.get seems like unsaved operations. Like so
case (Right(tagsProject), Right(tagsFiles)) =>
          Right(for {
            tagProject <- tagsProject
            tagFile <- tagsFiles
            if tagFile.commit_id == tagProject.commit.id
          } yield tagProject)
        case (_, Left(exception)) => Left(VersioningException(exception.message))
        case (Left(exception), _) => Left(VersioningException(exception.message))

Reply:
Changed to use pattern matching, thanks for the advice!